src/app/components/MediaLoader/Metadata/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useContext } from 'react';
+import { Helmet } from 'react-helmet';
+import filterForBlockType from '#app/lib/utilities/blockHandlers';
+import { RequestContext } from '#app/contexts/RequestContext';
+import {
+  ARTICLE_PAGE,
+  AV_EMBEDS,
+  CPS_ASSET,
+  MEDIA_ASSET_PAGE,
+  MEDIA_ARTICLE_PAGE,
+} from '#app/routes/utils/pageTypes';
+import { PageTypes } from '#app/models/types/global';
+import { AresMediaBlock, AresMediaMetadataBlock, MediaBlock } from '../types';
+
+const SUPPORTED_PAGE_TYPES = [
+  AV_EMBEDS,
+  ARTICLE_PAGE,
+  CPS_ASSET,
+  MEDIA_ASSET_PAGE,
+  MEDIA_ARTICLE_PAGE,
+] as PageTypes[];
+
+const getThumbnailUri = (url?: string) => {
+  if (!url) return null;
+
+  let imageUrl = url;
+
+  if (imageUrl.startsWith('http')) {
+    imageUrl = imageUrl.split('/')?.slice(2)?.join('/');
+  }
+
+  return `https://${imageUrl.replace('$recipe', '1024x576')}`;
+};
+
+const getUploadDate = (availableFrom?: string, firstPublished?: string) => {
+  const uploadDate = availableFrom || firstPublished;
+
+  if (!uploadDate) return null;
+
+  return new Date(uploadDate).toISOString();
+};
+
+type Props = {
+  blocks: MediaBlock[];
+  embedURL?: string;
+  embedded?: boolean;
+};
+
+const Metadata = ({ blocks, embedURL, embedded = false }: Props) => {
+  const { pageType } = useContext(RequestContext);
+
+  if (!SUPPORTED_PAGE_TYPES.includes(pageType)) return null;
+
+  const aresMediaBlock: AresMediaBlock = filterForBlockType(
+    blocks,
+    'aresMedia',
+  );
+
+  if (!aresMediaBlock) return null;
+
+  const { model: metadata }: AresMediaMetadataBlock =
+    filterForBlockType(aresMediaBlock?.model?.blocks, 'aresMediaMetadata') ??
+    {};
+
+  if (!metadata) return null;
+
+  const metadataJson = {
+    '@context': 'http://schema.org',
+    '@type': metadata?.format === 'audio' ? 'AudioObject' : 'VideoObject',
+    name: metadata?.title,
```

Comment:
All of these optional chaining will return `undefined`, related to the null question

Reply:
I lifted the logic from https://github.com/bbc/simorgh/blob/d03d06e0a0e3025b46c1546d297a3f4da93ff0e9/src/app/legacy/containers/MediaPlayer/helpers/metadata/index.js

`path` returns `undefined` by the looks of it. Do you think all the values should default to `undefined`?

Reply:
I guess it's more around consistency, seems a bit off using a mixture of undefined and null. It's probably more of an SEO question, should we either be omitting the the key and value if the value is falsy, or do we have them all undefined, or all of them null

Reply:
Its a good shout. I've pushed https://github.com/bbc/simorgh/pull/11940/commits/75338a4d1392df1d9dd3488a1a32dbf5712f2861 which should make the fallback values `null` by default and filter out any `null` values so that they aren't getting set in the JSON.